From: Isis Lovecruft <isis@torproject.org>
Subject: Remove SSL_OP_ALL from OpenSSL SSL_CTX_set_options().
Description: Add upstream patch which adds an OpenSSL option to disable SSLv2,
rather than SSL_OP_ALL | SSL_OP_NO_SSLv2.
Author:  Emanuele Giaquinta
Fixes: http://bugs.irssi.org/index.php?do=details&task_id=841
Bug: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=667014

--- a/src/core/network-openssl.c
+++ b/src/core/network-openssl.c
@@ -406,6 +406,7 @@
 		g_error("Could not allocate memory for SSL context");
 		return NULL;
 	}
+    SSL_CTX_set_options(ctx, SSL_OP_NO_SSLv2);
     
 	if (mycert && *mycert) {
 		char *scert = NULL, *spkey = NULL;
